src/librustc/mir/interpret/allocation.rs

```diff
@@ -35,6 +36,21 @@ pub struct Allocation<Tag=(),Extra=()> {
     pub extra: Extra,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
+pub struct AllocationBytes {
+    /// The actual bytes of the allocation.
+    /// Note that the bytes of a pointer represent the offset of the pointer
+    pub bytes: Vec<u8>,
+    /// Denotes undefined memory. Reading from undefined memory is forbidden in miri
```

Comment:
This is not true, `memcpy` of undefined memory is allowed. Also the important part is that it's forbidden *in Rust*.

I think we should state this positively: "Denotes which part of this allocation is initialized". Maybe it should even be renamed to `InitMask`?

Reply:
This was meant as 'interpreting undefined memory as a value of some type' is forbidden, i.e. not even integral values such as `u32` are allowed to be uninitialized. Can you propose a clearer alternative documentation?

Reply:
I did: "Denotes which part of this allocation is initialized". I wouldn't talk about what exact consequences that has as those rules are changing.

If you really want you can link to https://www.ralfj.de/blog/2019/07/14/uninit.html for an explanation why explicit tracking is needed. ;)

Reply:
Ah, so you would simply leave out mentioning the *effects* in miri as one should be able to infer that from the state. That sounds well, it just escaped me that this was your proposed replacement. Also read your blogpost, a (slightly unfortunate) difference of `undef` in llvm and here with `undef_mask`.

Reply:
Actually I think it is an unfortunate naming *alignment* because LLVM's `undef` is different in subtle ways from what we are doing. Which is why I propose to talk about the "initialized mask" instead.